zhaquirks/tuya/ts0601_valve.py

```diff
@@ -310,198 +311,169 @@ class ParksidePSBZS(EnchantedDevice):
-            attributes, manufacturer=foundation.ZCLHeader.NO_MANUFACTURER_ID
-        )
+    NoDelay = 0x00
+    TwentyFourHourDelay = 0x01
+    FortyEightHourDelay = 0x02
+    SeventyTwoHourDelay = 0x03
 
 
-class GiexValve(CustomDevice):
-    """GiEX valve device."""
+def giex_string_to_td(v: str) -> int:
+    """Convert Giex String Duration to seconds."""
+    dt = datetime.strptime(v, "%H:%M:%S,%f")
+    return timedelta(hours=dt.hour, minutes=dt.minute, seconds=dt.second).seconds
 
-    signature = {
-        MODELS_INFO: [
-            ("_TZE200_sh1btabb", "TS0601"),
-            ("_TZE200_a7sghmms", "TS0601"),
-            ("_TZE204_7ytb3h8u", "TS0601"),
-            ("_TZE200_7ytb3h8u", "TS0601"),
-        ],
-        ENDPOINTS: {
-            # <SimpleDescriptor endpoint=1 profile=260 device_type=0x0051
-            # input_clusters=[0x0000, 0x0004, 0x0005, 0xef00]
-            # output_clusters=[0x000a, 0x0019]>
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: zha.DeviceType.SMART_PLUG,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    GiexValveManufCluster.cluster_id,
-                ],
-                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
-            }
-        },
-    }
 
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                DEVICE_TYPE: zha.DeviceType.ON_OFF_SWITCH,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    TuyaOnOffNM,
-                    TuyaPowerConfigurationCluster,
-                    TuyaValveWaterConsumed,
-                    GiexValveManufCluster,
-                ],
-                OUTPUT_CLUSTERS: [Time.cluster_id, Ota.cluster_id],
-            }
-        }
-    }
+def giex_string_to_ts(v: str) -> int | None:
+    """Convert Giex String Duration datetime."""
+    dev_tz = timezone(timedelta(hours=4))
+    dev_dt = datetime.now(dev_tz)
+    try:
+        dt = datetime.strptime(v, "%H:%M:%S").replace(tzinfo=dev_tz)
+        dev_dt.replace(hour=dt.hour, minute=dt.minute, second=dt.second)
+    except ValueError:
+        return None  # on initial start the device will return '--:--:--'
+    return int(dev_dt.timestamp() + UNIX_EPOCH_TO_ZCL_EPOCH)
 
 
-class GiexValveVar02(CustomDevice):
-    """GiEX valve device, variant 2."""
+gx02_base_quirk = (
+    TuyaQuirkBuilder()
+    .tuya_battery(dp_id=108, power_cfg=TuyaPowerConfigurationCluster4AA)
+    .tuya_metering(dp_id=111)
+    .tuya_onoff(dp_id=2)
+    .tuya_number(
+        dp_id=103,
+        attribute_name="irrigation_num_times",
+        type=t.uint8_t,
+        min_value=0,
+        max_value=100,
+        step=1,
+        translation_key="irrigation_num_times",
+        fallback_name="Irrigation num Times",
```

Comment:
```suggestion
        fallback_name="Irrigation num times",
```
or is there a better name for this in general?

Reply:
Changed it to Irrigation cycles, think that's clearer

Reply:
Yep!